src/Data/AbstractAugmented.php

```diff
@@ -85,7 +85,13 @@ protected function getFromData($handle)
         $value = method_exists($this->data, 'value') ? $this->data->value($handle) : $this->data->get($handle);
 
         if (method_exists($this->data, 'getSupplement')) {
-            $value = $this->data->getSupplement($handle) ?? $value;
+            if (method_exists($this->data, 'hasSupplement')) {
+                $value = $this->data->hasSupplement($handle)
+                    ? $this->data->getSupplement($handle)
+                    : $value;
+            } else {
+                $value = $this->data->getSupplement($handle) ?? $value;
+            }
```

Comment:
The getSupplement check was enough IMO. If you have one method you're more than likely using the trait and would have both.

Reply:
I suspected that. Reverting!